prediction_market_agent/tools/mech/utils.py

```diff
@@ -144,4 +155,4 @@ def mech_request_local(question: str, mech_tool: MechTool) -> OutcomePrediction:
         raise ValueError(f"Mech type '{mech_tool}' not supported")
 
     result = completion_str_to_json(str(response[0]))
-    return OutcomePrediction.model_validate(result)
+    return MechResponse.model_validate(result)
```

Comment:
How is the structure defined i.e. how do you enforce the output format to be `MechResponse`-like?

Reply:
It's just defined in the system prompt for every mech, e.g. https://github.com/valory-xyz/mech/blob/main/packages/valory/customs/prediction_request/prediction_request.py#L274

Not a strong guarantee, but I've never seen it get it wrong.